plugin/backend_linux.go

```diff
@@ -294,11 +294,11 @@ func (pm *Manager) Remove(name string, config *types.PluginRmConfig) error {
 	}
 
 	id := p.GetID()
-	pm.pluginStore.Remove(p)
 	pluginDir := filepath.Join(pm.libRoot, id)
 	if err := os.RemoveAll(pluginDir); err != nil {
-		logrus.Warnf("unable to remove %q from plugin remove: %v", pluginDir, err)
+		return fmt.Errorf("unable to remove %q from plugin remove: %v", pluginDir, err)
```

Comment:
It's provided through this package; https://github.com/pkg/errors#adding-context-to-an-error